setup.py

```diff
@@ -112,8 +112,20 @@ def build_extensions(self):
             VERSION = line.strip().split()[-1][1:-1]
             break
 
-with open('requirements/default.txt') as fid:
-    INSTALL_REQUIRES = [l.strip() for l in fid.readlines() if l]
+
+def parse_requirements_file(filename):
+    with open(filename) as fid:
+        requires = [l.strip() for l in fid.readlines() if l]
+
+    return requires
+
+
+INSTALL_REQUIRES = parse_requirements_file('requirements/default.txt')
+extras_require = {
+    'docs': parse_requirements_file('requirements/docs.txt'),
+    'optional': parse_requirements_file('requirements/optional.txt'),
+    'test': parse_requirements_file('requirements/test.txt'),
```

Comment:
i'm not too sure about this test dependency.

Reply:
Same can be said for `docs` :). I'm fine with the current version.